src/common/serialization.rs

```diff
@@ -146,6 +156,83 @@ impl FileAttrSimple {
             blksize: 0,
         }
     }
+
+    pub fn tostat(&self, statbuf: &mut [u8]) {
```

Comment:
How about to convert `FileAttrSimple` into `stat` in an `into()` function and just convert `stat` into `[u8]` in this function?

Reply:
I think this reduces one copy

Reply:
You are right.